src/pages/index.js

```diff
@@ -112,22 +182,79 @@ function closeModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+
+  const cardSubmitButton = evt.submitter;
+  setButtonText(cardSubmitButton, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitButton, false);
+    });
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
-    name: addCardNameInput.value,
-    link: addCardLinkInput.value,
-  };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(addCardModal);
+  setButtonText(cardSubmitButton, true);
+
+  api
+    .addCard({
+      name: addCardNameInput.value,
+      link: addCardLinkInput.value,
+    })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      disableButton(cardSubmitButton, settings);
+      closeModal(addCardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitButton, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  console.log("Avatar URL being submitted:", avatarInput.value);
+  setButtonText(avatarSubmitButton, true);
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      console.log("Response from server:", data);
+      profileAvatar.src = data.avatar;
+      evt.target.reset();
```

Comment:
* Please, disable the `avatar` submit button in `.then` after submission as well https://snipboard.io/IPoaZm.jpg 

```js

 disableButton(avatarSubmitButton, settings);
```